live/client.py

```diff
@@ -91,11 +126,24 @@ def start_event(self, event_id):
             headers = self.generate_headers_with_authentication_enabled(
                 url, self.user, self.api_key)
 
-        # Send request
-        response = requests.request(method='POST', url=url, data=body,
-                                    headers=headers)
-
-        return response
+        # Send request and do exception handling
+        try:
+            response = requests.request(method='POST', url=url, data=body,
+                                        headers=headers)
+            if response.status_code != 200:
+                raise InvalidHTTPResponse(f"Start Event {event_id} Failed \n"
+                                          f"Please Check If You Passed "
+                                          f"The Correct Event Id \n"
+                                          f"Or If This Event is allowed to "
+                                          f"Start")
+        except requests.exceptions.RequestException as e:
+            print(e)
+            return
+        except InvalidHTTPResponse as e:
+            print(e)
+            return
```

Comment:
Seems like `start_event` would fail silently if the request fails. I would advise to raise an exception for the caller if the request fails. 

From the point of of the caller the code would look like this:

```python
try:
    element_client.start_event('123')
except InvalidHTTPResponse:
    logging.exception("Fail to start event")
```

For that to be the case, you need to handle both `requests` exceptions and an invalid status code, like you're doing now. It would look something like this:

```python
        try:
            response = requests.post(url=url, data=body, headers=headers)
        except requests.exceptions.RequestException as e:
            raise InvalidHTTPResponse(f"Fail to send request to start event {e}")
        if response.status_code != 200:
            raise InvalidHTTPResponse(f"Start Event {event_id} Failed \n"
                                      f"Please Check If You Passed "
                                      f"The Correct Event Id \n"
                                      f"Or If This Event is allowed to "
                                      f"Start. Response: {response.status_code}: {response.text}")
```